webcodecs/video-track-reader.html

```diff
@@ -22,52 +22,52 @@
   return video.captureStream();
 }
 
-async_test(async function(t) {
+promise_test(async function(t) {
   let stream = await getMediaStream();
   let vtr = new VideoTrackReader(stream.getVideoTracks()[0]);
-
-  t.done();
 }, 'Test we can construct a VideoTrackReader.');
 
-async_test(async function(t) {
+promise_test(async function(t) {
   let stream = await getMediaStream();
   let vtr = new VideoTrackReader(stream.getVideoTracks()[0]);
 
   let numberFrames = 0;
 
-  vtr.start(t.step_func((frame) => {
-    assert_equals(frame.codedWidth, testVideo.width);
-    assert_equals(frame.codedHeight, testVideo.height);
-    assert_not_equals(frame.timestamp, null);
-    frame.destroy();
-
-    if (++numberFrames == 5){
-      vtr.stop();
-      t.done();
-    }
-  }));
-
+  return new Promise(resolve => {
+    vtr.start(t.step_func(frame => {
+      assert_equals(frame.codedWidth, testVideo.width);
+      assert_equals(frame.codedHeight, testVideo.height);
+      assert_not_equals(frame.timestamp, null);
+      frame.destroy();
+
+      if (++numberFrames == 5) {
+        vtr.stop();
+        resolve();
+      }
+    }));
+  });
 }, 'Test we can start and stop a VideoTrackReader.');
 
-async_test(async function(t) {
+promise_test(async function(t) {
   let stream = await getMediaStream();
   let vtr = new VideoTrackReader(stream.getVideoTracks()[0]);
 
   let stoppedOnce = false;
 
-  vtr.start(t.step_func(() => {
-    if(!stoppedOnce) {
-      vtr.stop();
-      vtr.start(t.step_func_done());
-      stoppedOnce = true;
-    } else {
-      assert_unreached("A stopped callback should never be called again");
-    }
-  }));
-
+  return new Promise(resolve => {
+    vtr.start(t.step_func(() => {
+      if(!stoppedOnce) {
```

Comment:
This could be simplified to an assert, no branching.

Reply:
Done.